nilearn/signal.py

```diff
@@ -594,11 +608,21 @@ def clean(signals, runs=None, detrend=True, standardize='zscore',
         signals = _standardize(signals, standardize=False, detrend=detrend)
         if confounds is not None:
             confounds = _standardize(confounds, standardize=False,
-                                detrend=detrend)
-    if use_filter:
-        # check if filter parameters are satisfied and filter according to the strategy
-        signals, confounds = _filter_signal(signals, confounds, filter,
-                                            low_pass, high_pass, t_r)
+                                     detrend=detrend)
+
+    if filter_type == 'butterworth':
+        signals = butterworth(signals, sampling_rate=1. / t_r,
+                              low_pass=low_pass, high_pass=high_pass)
+        if confounds is not None:
+            # Apply low- and high-pass filters to keep filters orthogonal
+            # (according to Lindquist et al. (2018))
+            confounds = butterworth(confounds, sampling_rate=1. / t_r,
+                                    low_pass=low_pass, high_pass=high_pass)
+        # apply sample_mask to remove censored volumes after signal filtering
+        if sample_mask is not None:
```

Comment:
I'm not sure I understand the logic here: if there is an outlying value, then signals and confounds estimates will be (possibly strongly) biased. I think that the values out of samples mask should be soehow imputed so that they do not contaminate the others ?

Reply:
Yes you are correct here. The steps should be:
1. censor volumes flagged by `sample_mask`
2. impute volumes flagged by `sample_mask`
3. apply butterworth filter
4. censor volumes flagged by `sample_mask`

I haven't implement any thing related to interpolation and was planning on doing it in a different PR. If you think it makes more sense to address it here, I can convert this PR to draft and do it in one PR. WDYT?

Reply:
I propose to take care of it in this PR. Thx !